homeassistant/components/econet/climate.py

```diff
@@ -0,0 +1,243 @@
+        return self.thermostat.set_point
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self.thermostat.humidity
+
+    @property
+    def target_humidity(self):
+        """Return the humidity we try to reach."""
+        if self.thermostat.supports_humidifier:
+            return self.thermostat.dehumidifier_set_point
+        return None
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if self.hvac_mode == HVAC_MODE_COOL:
+            return self.thermostat.cool_set_point
+        if self.hvac_mode == HVAC_MODE_HEAT:
+            return self.thermostat.heat_set_point
+        return None
+
+    @property
+    def target_temperature_low(self):
+        """Return the lower bound temperature we try to reach."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            return self.thermostat.heat_set_point
+        return None
+
+    @property
+    def target_temperature_high(self):
+        """Return the higher bound temperature we try to reach."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            return self.thermostat.cool_set_point
+        return None
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        target_temp = kwargs.get(ATTR_TEMPERATURE)
+        target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+        target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        if target_temp:
+            self.thermostat.set_set_point(target_temp, None, None)
+            return
+        if target_temp_low or target_temp_high:
+            self.thermostat.set_set_point(None, target_temp_high, target_temp_low)
+            return
+
+    @property
+    def is_aux_heat(self):
+        """Return true if aux heater."""
+        return self.thermostat.mode == ThermostatOperationMode.EMERGENCY_HEAT
+
+    @property
+    def hvac_modes(self):
+        """Return hvac operation ie. heat, cool mode.
+
+        Needs to be one of HVAC_MODE_*.
+        """
+        econet_modes = self.thermostat.modes
+        op_list = []
+        for mode in econet_modes:
+            if mode not in [
+                ThermostatOperationMode.UNKNOWN,
+                ThermostatOperationMode.EMERGENCY_HEAT,
+            ]:
+                ha_mode = ECONET_STATE_TO_HA[mode]
+                op_list.append(ha_mode)
+        return op_list
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool, mode.
+
+        Needs to be one of HVAC_MODE_*.
+        """
+        econet_mode = self.thermostat.mode
+        _current_op = HVAC_MODE_OFF
+        if econet_mode is not None:
+            _current_op = ECONET_STATE_TO_HA[econet_mode]
+
+        return _current_op
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        hvac_mode_to_set = HA_STATE_TO_ECONET.get(hvac_mode)
+        self.thermostat.set_mode(hvac_mode_to_set)
```

Comment:
Does this call handle a None argument for `hvac_mode_to_set` parameter?

Reply:
It won't crash or anything, but it won't work.

Reply:
Ok, then we should check for that and raise a ValueError with an appropriate message as exception argument.

Reply:
Done.